hibernate-ehcache/src/main/java/org/hibernate/cache/ehcache/internal/HibernateEhcacheUtils.java

```diff
@@ -62,6 +67,29 @@ public static Configuration loadAndCorrectConfiguration(URL url) {
 		return config;
 	}
 
+	static void setCacheManagerNameIfNeeded(SessionFactoryOptions settings, Configuration configuration, Map properties) {
+		overwriteCacheManagerIfConfigured( configuration, properties );
+		if (configuration.getName() == null) {
+			String sessionFactoryName = settings.getSessionFactoryName();
+			if (sessionFactoryName != null) {
+				configuration.setName( sessionFactoryName );
+			}
+			else {
+				configuration.setName( "Hibernate " + UUID.randomUUID().toString() );
```

Comment:
IMO it would be better to use the SessionFactory's UUID.  However, currently this code does not have access to that because only the `SessionFactoryOptions` are passed, not the `SessionFactoryImplementor`.

@dreab8 @Naros what do you think about either:

1. Managing the SF UUID on `SessionFactoryOption` instead (pro - wider usage/availability; con -?)
2. Passing the UUID to the `RegionFactory` start up.
3. Passing the `SessionFactoryImplementor` to the `RegionFactory` start up.

(3) is my least favorite

Reply:
https://hibernate.atlassian.net/browse/HHH-12440

Reply:
Unless we decide to drop `hibernate-ehcache` per other discussion

Reply:
@alexsnaps @henri-tremblay @chrisdennis What do you guys think of using the SessionFactory's UUID instead?  Worth the (small) effort?

Reply:
It's nice to match UUIDs. If it seems useful for other purposes, why not. Otherwise, it's probably not essential

Reply:
Right, that was my thought.  It might help in other places and we can still expose it from SessionFactory.. so why not